test/e2e/mock-response-data/snaps/snap-binary-mocks.ts

```diff
@@ -0,0 +1,328 @@
+  'test/e2e/mock-response-data/snaps/snap-binaries-and-headers';
+const NPM_REGISTRY_METAMASK_BASE_URL = 'https://registry.npmjs.org/@metamask';
+
+/**
+ * Sorts an array of file names by version strings semantically (e.g., 1.2.0, 1.10.0, 2.0.0).
+ * Modifies the array in place.
+ *
+ * @param versions - An array of version strings to sort.
+ */
+function sortFilesByVersion(versions: string[]): void {
+  versions.sort((a, b) => {
+    const partsA = a.split('.').map(Number);
+    const partsB = b.split('.').map(Number);
+    for (let i = 0; i < Math.max(partsA.length, partsB.length); i++) {
+      const valA = partsA[i] ?? 0;
+      const valB = partsB[i] ?? 0;
+      if (valA < valB) {
+        return -1;
+      }
+      if (valA > valB) {
+        return 1;
+      }
+    }
+    return 0;
+  });
+}
+
+/**
+ * Retrieves the latest version of a snap from the local file system based on its name prefix.
+ * It looks for files named `<snapNamePrefix>-<version>.txt`, collects all versions,
+ * sorts them semantically, and returns the latest one.
+ *
+ * @param snapNamePrefix - The prefix of the snap name (e.g., 'bip32-example-snap').
+ * @returns The latest version string found for the snap.
+ * @throws If no version file is found for the given snap prefix in the directory.
+ */
+export function getLocalSnapLatestVersion(snapNamePrefix: string): string {
+  const files = fs.readdirSync(SNAP_ASSETS_RELATIVE_PATH);
+  const sanitizedSnapNamePrefix = escapeRegExp(snapNamePrefix);
+  const versionRegex = new RegExp(
+    `${sanitizedSnapNamePrefix}-(\\d+\\.\\d+\\.\\d+)\\.txt`,
+    'u',
+  );
+  const foundVersions: string[] = [];
+  for (const file of files) {
+    const match = file.match(versionRegex);
+    if (match?.[1]) {
+      foundVersions.push(match[1]);
+    }
+  }
+
+  if (foundVersions.length === 0) {
+    throw new Error(
+      `No version found for snap "${snapNamePrefix}" in directory "${SNAP_ASSETS_RELATIVE_PATH}".`,
+    );
+  }
+
+  sortFilesByVersion(foundVersions);
+
+  // The last element after sorting will be the latest version
+  return foundVersions[foundVersions.length - 1];
+}
+
+/**
+ * Creates a mock endpoint for a snap hosted on the npm registry.
+ * It reads the snap binary and headers from local files based on the snap name prefix and version.
+ * The version can be explicitly provided or determined by scanning local files.
+ *
+ * @param options - The options for creating the snap mock.
+ * @param options.mockServer - The mockttp server instance.
+ * @param options.snapNamePrefix - The prefix of the snap name (e.g., 'bip32-example-snap').
+ * @param [options.specificVersion] - An optional specific version string for the snap. If not provided, `getLocalSnapLatestVersion` will be used to determine the version.
+ * @returns A promise that resolves to the mocked endpoint.
+ */
+async function createSnapMock(options: {
+  mockServer: Mockttp;
+  snapNamePrefix: string;
+  specificVersion?: string;
```

Comment:
we'll use the latest version of the snap by default, but we allow to pass an optional version, if we want to use a specific version (like in the case of upgrade flow)